Replace HashMap with BTreeMap in storage

kpp opened the pull request:
# Description
This may probably affect cycle counts

## Linked Issues
- Related to https://github.com/chainwayxyz/citrea/issues/945

eyusufatik commented:
I need benchmarks for this both in zk (which is probably better) and for native execution.

kpp commented:
Before:

```
2024-08-29T10:37:34.663863Z  INFO citrea_risc0_bonsai_adapter::host: Added hint to guest with size 129482
Running sequencer commitments in DA slot
going into apply_soft_confirmations_from_sequencer_commitments
out of apply_soft_confirmations_from_sequencer_commitments
2024-08-29T10:37:50.372488Z  INFO executor: risc0_zkvm::host::server::exec::executor: execution time: 13.772917002s
2024-08-29T10:37:50.372542Z  INFO executor: risc0_zkvm::host::server::session: number of segments: 36
2024-08-29T10:37:50.372554Z  INFO executor: risc0_zkvm::host::server::session: total cycles: 37748736
2024-08-29T10:37:50.372564Z  INFO executor: risc0_zkvm::host::server::session: user cycles: 24851773
```

After:

```
2024-08-29T10:49:41.680428Z  INFO citrea_risc0_bonsai_adapter::host: Added hint to guest with size 131298
Running sequencer commitments in DA slot
going into apply_soft_confirmations_from_sequencer_commitments
out of apply_soft_confirmations_from_sequencer_commitments
2024-08-29T10:49:57.389553Z  INFO executor: risc0_zkvm::host::server::exec::executor: execution time: 13.76639099s
2024-08-29T10:49:57.389604Z  INFO executor: risc0_zkvm::host::server::session: number of segments: 36
2024-08-29T10:49:57.389616Z  INFO executor: risc0_zkvm::host::server::session: total cycles: 37224448
2024-08-29T10:49:57.389626Z  INFO executor: risc0_zkvm::host::server::session: user cycles: 24589729
```

kpp commented:
```
        for i in 0..10 {
            for _ in 0..5 {
                let _ = sequencer.client.send_eth(to, None, None, None, 1u128).await;
            }
            sequencer.client.send_publish_batch_request().await;
            sequencer.wait_for_l2_height(seq_height0 + i + 1, None).await;
        }
```


Before:

```
2024-08-29T13:06:04.029375Z  INFO citrea_risc0_bonsai_adapter::host: Added hint to guest with size 215674
Running sequencer commitments in DA slot
going into apply_soft_confirmations_from_sequencer_commitments
out of apply_soft_confirmations_from_sequencer_commitments
2024-08-29T13:07:55.164017Z  INFO executor: risc0_zkvm::host::server::exec::executor: execution time: 108.700436027s
2024-08-29T13:07:55.164091Z  INFO executor: risc0_zkvm::host::server::session: number of segments: 196
2024-08-29T13:07:55.164106Z  INFO executor: risc0_zkvm::host::server::session: total cycles: 204734464
2024-08-29T13:07:55.164129Z  INFO executor: risc0_zkvm::host::server::session: user cycles: 151283298
```

After:

```
2024-08-29T13:12:29.137427Z  INFO citrea_risc0_bonsai_adapter::host: Added hint to guest with size 214444
Running sequencer commitments in DA slot
going into apply_soft_confirmations_from_sequencer_commitments
out of apply_soft_confirmations_from_sequencer_commitments
2024-08-29T13:14:20.776245Z  INFO executor: risc0_zkvm::host::server::exec::executor: execution time: 109.512394883s
2024-08-29T13:14:20.776331Z  INFO executor: risc0_zkvm::host::server::session: number of segments: 194
2024-08-29T13:14:20.776346Z  INFO executor: risc0_zkvm::host::server::session: total cycles: 203423744
2024-08-29T13:14:20.776377Z  INFO executor: risc0_zkvm::host::server::session: user cycles: 149839975
```
